socketcan/libcanard/src/socketcan.c

```diff
@@ -244,22 +245,24 @@ int16_t socketcanPop(const SocketCANFD       fd,
 
         // Obtain the CAN frame time stamp from the kernel.
         // This time stamp is from the CLOCK_REALTIME kernel source.
-        const struct cmsghdr* cmsg = CMSG_FIRSTHDR(&msg);
-        struct timeval        tv   = {0};
-        assert(cmsg != NULL);
-        if (cmsg->cmsg_level == SOL_SOCKET && cmsg->cmsg_type == SO_TIMESTAMP)
-        {
-            (void) memcpy(&tv, CMSG_DATA(cmsg), sizeof(tv));  // Copy to avoid alignment problems
-            assert(tv.tv_sec >= 0 && tv.tv_usec >= 0);
+        if(out_timestamp_usec){
```

Comment:
1. Implicit cast to boolean is prohibited by most high-integrity coding standards.
2. Please run clang-format: `clang-format src/*.[ch]` or something similar.